### PR TITLE
fetches userId from the oauth_token (the part before '-') if manually…

### PR DESCRIPTION
**Pull request by webstersx**
… specified -- user_id isn't in Twitter's own instructions for implementing twitter login: https://dev.twitter.com/web/sign-in/implementing

### CODE REVIEW

**Comment by webstersx** [comment]
oops, typo.

should read as: if **not** manually specified

**Comment by ghaiklor** [comment]
@webstersx could you cover it with tests ?

And, does it make sense ? I mean, `oauth-token` doesn't contain `user_id` so why you need this ?

**Comment by webstersx** [comment]
@ghaiklor my twitter 'oauth_token' contains my twitter user id

Am I just misunderstanding that the user_id is mean to be a local user in passport-twitter-token?

when I don't specify user_id in the body, passport causes a 500: InternalOAuthError: failed to fetch user profile

**Comment by webstersx** [comment]
It looks as though when _skipExtendedUserProfile = false, A twitter user id is required as 'user_id'

Since the user_id is already in the oauth_token, my thinking is that it would make sense to just read it out of there instead of requiring another param

``` javascript
TwitterTokenStrategy.prototype.userProfile = function(token, tokenSecret, params, done) {
  if (!this._skipExtendedUserProfile) {
    this._oauth.get('https://api.twitter.com/1.1/users/show.json?user_id=' + params.user_id, token, tokenSecret, function (err, body, res) {
```

As long as the format of the oauth_token is always of the format {twitter user id}-{random characters} this should be okay -- happy to hear your thoughts

**Comment by ghaiklor** [comment]
@webstersx I'll be happy to merge this, but as I ask, cover it with tests, please.
Your code doesn't break backward compatibility, but it must be covered with tests.

File: src/index.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ export default class TwitterTokenStrategy extends OAuthStrategy {
 
     let token = (req.body && req.body[this._oauthTokenField]) || (req.query && req.query[this._oauthTokenField]);
     let tokenSecret = (req.body && req.body[this._oauthTokenSecretField]) || (req.query && req.query[this._oauthTokenSecretField]);
-    let userId = (req.body && req.body[this._userIdField]) || (req.query && req.query[this._userIdField]);
+    let userId = (req.body && req.body[this._userIdField]) || (req.query && req.query[this._userIdField]) || (token.substr(0, token.indexOf('-')));
```
Comment (ghaiklor) [inline src/index.js]: You can change to `token.split(`-`)[0]`. It more easily to read what it did.

**Comment by webstersx** [comment]
@ghaiklor I'm having trouble writing the tests as I don't see where the existing userId portion is being tested :( (I'm also an iOS programmer by trade so there's that too!)

Can you offer any suggestions for how to accomplish this?

Cheers

**Comment by ghaiklor** [comment]
@webstersx I can add it myself though, but tomorrow. I will try not to forgot about this :+1:

**Comment by ghaiklor** [comment]
@webstersx published under 1.1.0

**Pull request merged**